arkouda/pdarrayclass.py

```diff
@@ -2341,6 +2342,82 @@ def corr(x: pdarray, y: pdarray) -> np.float64:
     return parse_single_value(generic_msg(cmd="corr", args={"x": x, "y": y}))
 
 
+@typechecked
+def divmod(
+    x: Union[numeric_scalars, pdarray],
+    y: Union[numeric_scalars, pdarray],
+    where: Union[bool, pdarray] = True,
+) -> Tuple[pdarray, pdarray]:
+    """
+    Parameters
+    ----------
+    x : numeric_scalars(float_scalars, int_scalars) or pdarray
+        The values that will be the numerator of the division, the dividends
+    y : numeric_scalars(float_scalars, int_scalars) or pdarray
+        The values that will be the denominator of the division, the divisors
+    where : Boolean or pdarray
+        This condition is broadcast over the input. At locations where the condition is True, the
+        corresponding value will be divided using floor and modular division. Elsewhere, it will retain
+        its original value. Default set to True.
+
+    Returns
+    -------
+    (pdarray, pdarray)
+        Returns a tuple that contains quotient and remainder of the division
+
+    Raises
+    ------
+    TypeError
+        At least one entry must be a pdarray
+    ValueError
+        If both inputs are both pdarrays, their size must match
+    ZeroDivisionError
+        No entry in y is allowed to be 0, to prevent division by zero
+
+    Notes
+    -----
+    The div is calculated by x // y
+    The mod is calculated by x % y
+
+    Examples
+    --------
+    >>> x = ak.arange(5, 10)
+    >>> y = ak.array([2, 1, 4, 5, 8])
+    >>> ak.divmod(x,y)
+    (array([2 6 1 1 1]), array([1 0 3 3 1]))
+    >>> ak.divmod(x,y, x % 2 == 0)
+    (array([5 6 7 1 9]), array([5 0 7 3 9]))
+    """
+    from arkouda.numeric import cast as akcast
+    from arkouda.numeric import where as akwhere
+    from arkouda.pdarraycreation import full
+
+    if not isinstance(x, pdarray) and not isinstance(y, pdarray):
+        raise TypeError("At least one entry must be a pdarray.")
+
+    if isinstance(x, pdarray) and isinstance(y, pdarray):
+        if x.size != y.size:
+            raise ValueError(f"size mismatch {x.size} {y.size}")
+
+    if (not isinstance(y, pdarray) and y == 0) or (isinstance(y, pdarray) and not all(y)):
+        raise ZeroDivisionError("Can not divide by zero")
```

Comment:
This line is what is preventing you from doing `divmod` on `uint` arrays. There's no `ak.all` reduction defined for `uint`. If you change the `not all(y)` to `any(y == 0)` then there's no reason you can't do `divmod(uint_pda, uint_pda)`


I think the cleanest way to do this is something like
```python
equal_zero = y == 0
if equal_zero if isinstance(equal_zero, bool) else any(equal_zero):
```

```python
>>> x = ak.arange(1, 11, dtype=ak.uint64)
>>> y = ak.ones(10, dtype=ak.uint64)
>>> x
array([1 2 3 4 5 6 7 8 9 10])

>>> y
array([1 1 1 1 1 1 1 1 1 1])

>>> x.dtype
dtype('uint64')

>>> y.dtype
dtype('uint64')

>>> ak.divmod(x,y)
(array([1 2 3 4 5 6 7 8 9 10]), array([0 0 0 0 0 0 0 0 0 0]))

>>> ak.divmod(y,x)
(array([1 0 0 0 0 0 0 0 0 0]), array([0 1 1 1 1 1 1 1 1 1]))
```